components/builder-web/app/shared/date/date.component.ts

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2022 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Component, Input } from '@angular/core';
+
+import * as moment from 'moment';
+
+@Component({
+  selector: 'hab-date',
+  template: `<mat-label [matTooltip]="tooltip" matTooltipPosition="above">{{ displayDate }}</mat-label>`
+})
+export class DateComponent {
+  @Input() date: string = '';
+
+  get tooltip() {
+    let tip;
+
+    if (this.date && this.date.trim() !== '') {
+      tip = this.date;
+    }
+
+    return moment(this.date).format('ddd DD-MMM-YYYY, hh:mm A');
+  }
+
+  get displayDate() {
+    return moment(this.date).fromNow();
```

Comment:
Can we make sure the time comparison is purely GMT based? The date is GMT but Now is the local time zone which gives an inaccurate time.

Reply:
Moment offers various parsing options. moment(date) will parse the date in the context of user's local time. There is another variant monent.utc(date) to part as a UTC date. I think we can use this.

https://momentjs.com/guides/#/parsing/